From 74145efe1a07e61de9ab6a14720e7cd8d92a1f3f Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 14 Dec 2012 08:12:48 +0100
Subject: [PATCH 1/2] Move installation path definitions into a separate
 header file

This is to recompile only few files when installation paths are changed;
currently all files are recompiled.

Fix issue #324.
---
 CMakeLists.txt                                     |    5 ++++
 oce_build_config.h.cmake                           |   17 ------------
 oce_build_paths.h.cmake                            |   27 ++++++++++++++++++++
 src/Aspect/Aspect_Driver.cxx                       |    2 ++
 src/DPrsStd/DPrsStd.cxx                            |    2 ++
 src/Draw/Draw.cxx                                  |    2 ++
 src/Draw/Draw_PloadCommands.cxx                    |    2 ++
 src/Graphic3d/Graphic3d_GraphicDevice.cxx          |    2 ++
 src/Graphic3d/Graphic3d_Texture1D.cxx              |    2 ++
 src/Graphic3d/Graphic3d_Texture2D.cxx              |    2 ++
 src/Graphic3d/Graphic3d_TextureEnv.cxx             |    2 ++
 src/Graphic3d/Graphic3d_WNTGraphicDevice.cxx       |    2 ++
 src/Plugin/Plugin.cxx                              |    2 ++
 src/Resource/Resource_Manager.cxx                  |    2 ++
 src/UnitsAPI/UnitsAPI.cxx                          |    2 ++
 src/ViewerTest/ViewerTest.cxx                      |    3 +++
 .../XmlLDrivers_DocumentStorageDriver.cxx          |    2 ++
 17 files changed, 61 insertions(+), 17 deletions(-)
 create mode 100644 oce_build_paths.h.cmake

From 9e17cd4a8ebfeed47de4dd677df31254690ab685 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Tue, 18 Dec 2012 07:39:59 +0100
Subject: [PATCH 2/2] Protect oce-paths.h inclusion

This may be needed when compiling with WOK.
---
 CMakeLists.txt                                        |    1 +
 src/Aspect/Aspect_Driver.cxx                          |    4 +++-
 src/DPrsStd/DPrsStd.cxx                               |    4 +++-
 src/Draw/Draw.cxx                                     |    4 +++-
 src/Draw/Draw_PloadCommands.cxx                       |    4 +++-
 src/Graphic3d/Graphic3d_GraphicDevice.cxx             |    4 +++-
 src/Graphic3d/Graphic3d_Texture1D.cxx                 |    4 +++-
 src/Graphic3d/Graphic3d_Texture2D.cxx                 |    4 +++-
 src/Graphic3d/Graphic3d_TextureEnv.cxx                |    4 +++-
 src/Graphic3d/Graphic3d_WNTGraphicDevice.cxx          |    4 +++-
 src/Plugin/Plugin.cxx                                 |    4 +++-
 src/Resource/Resource_Manager.cxx                     |    4 +++-
 src/UnitsAPI/UnitsAPI.cxx                             |    4 +++-
 src/ViewerTest/ViewerTest.cxx                         |    4 +++-
 src/XmlLDrivers/XmlLDrivers_DocumentStorageDriver.cxx |    4 +++-
 15 files changed, 43 insertions(+), 14 deletions(-)

Index: oce/src/DPrsStd/DPrsStd.cxx
===================================================================
--- oce.orig/src/DPrsStd/DPrsStd.cxx
+++ oce/src/DPrsStd/DPrsStd.cxx
@@ -37,6 +37,10 @@
 #include <OSD_Directory.hxx>
 #include <OSD_File.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
 #pragma warning(4:4190)
Index: oce/src/Draw/Draw.cxx
===================================================================
--- oce.orig/src/Draw/Draw.cxx
+++ oce/src/Draw/Draw.cxx
@@ -71,6 +71,10 @@
 
 #include <stdio.h>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 extern Standard_Boolean Draw_ParseFailed;
 
 Standard_EXPORT Draw_Viewer dout;
Index: oce/src/Draw/Draw_PloadCommands.cxx
===================================================================
--- oce.orig/src/Draw/Draw_PloadCommands.cxx
+++ oce/src/Draw/Draw_PloadCommands.cxx
@@ -28,6 +28,10 @@
 #include <Draw_MapOfAsciiString.hxx>
 #include <Draw.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 static Handle(Resource_Manager) myResources;
 
 //=======================================================================
Index: oce/src/Plugin/Plugin.cxx
===================================================================
--- oce.orig/src/Plugin/Plugin.cxx
+++ oce/src/Plugin/Plugin.cxx
@@ -28,6 +28,10 @@
 
 #include <TCollection_AsciiString.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 static Standard_Character tc[1000];
 static Standard_PCharacter thePluginId = tc;
 
Index: oce/src/Resource/Resource_Manager.cxx
===================================================================
--- oce.orig/src/Resource/Resource_Manager.cxx
+++ oce/src/Resource/Resource_Manager.cxx
@@ -34,6 +34,10 @@
 
 #include <errno.h>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 #define END      0
 #define EMPTY    1
 #define COMMENT  2
Index: oce/src/UnitsAPI/UnitsAPI.cxx
===================================================================
--- oce.orig/src/UnitsAPI/UnitsAPI.cxx
+++ oce/src/UnitsAPI/UnitsAPI.cxx
@@ -26,6 +26,10 @@
 #include <Standard_NoSuchObject.hxx>
 #include <OSD_Environment.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 static Handle(Resource_Manager) CurrentUnits,SICurrentUnits,MDTVCurrentUnits;
 static Units_UnitsSystem LocalSystemUnits,SILocalSystemUnits,MDTVLocalSystemUnits;
 static TCollection_AsciiString rstring;
Index: oce/src/ViewerTest/ViewerTest.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest.cxx
+++ oce/src/ViewerTest/ViewerTest.cxx
@@ -81,6 +81,11 @@
 
 #include <NIS_InteractiveContext.hxx>
 #include <NIS_Triangulated.hxx>
+
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 extern int ViewerMainLoop(Standard_Integer argc, const char** argv);
 
 #include <Quantity_Color.hxx>
Index: oce/src/XmlLDrivers/XmlLDrivers_DocumentStorageDriver.cxx
===================================================================
--- oce.orig/src/XmlLDrivers/XmlLDrivers_DocumentStorageDriver.cxx
+++ oce/src/XmlLDrivers/XmlLDrivers_DocumentStorageDriver.cxx
@@ -49,6 +49,10 @@
 #include <OSD_File.hxx>
 #include <OSD_Environment.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 #define STORAGE_VERSION      "STORAGE_VERSION: "
 #define REFERENCE_COUNTER    "REFERENCE_COUNTER: "
 #define MODIFICATION_COUNTER "MODIFICATION_COUNTER: "
